packages/hoppscotch-app/src/helpers/editor/extensions/HoppEnvironment.ts

```diff
@@ -58,27 +59,43 @@ const cursorTooltipField = (aggregateEnvs: AggregateEnvironment[]) =>
       )
         return null
 
-      const envName =
-        aggregateEnvs.find(
-          (env) => env.key === text.slice(start - from, end - from)
-          // env.key === word.slice(wordSelection.from + 2, wordSelection.to - 2)
-        )?.sourceEnv ?? "Choose an Environment"
+      const parsedEnvKey = text.slice(start - from, end - from)
 
-      const envValue =
-        aggregateEnvs.find(
-          (env) => env.key === text.slice(start - from, end - from)
-          // env.key === word.slice(wordSelection.from + 2, wordSelection.to - 2)
-        )?.value ?? "Not found"
+      const tooltipEnv = aggregateEnvs.find(
+        (env) => env.key === parsedEnvKey
+        // env.key === word.slice(wordSelection.from + 2, wordSelection.to - 2)
+      )
+
+      const envName = tooltipEnv?.sourceEnv ?? "Choose an Environment"
+
+      const envValue = tooltipEnv?.value ?? "Not found"
 
       const result = parseTemplateStringE(envValue, aggregateEnvs)
 
       const finalEnv = E.isLeft(result) ? "error" : result.right
 
       const selectedEnvType = getSelectedEnvironmentType()
 
-      const envTypeIcon = `<i class="inline-flex items-center pr-2 mr-2 -my-1 text-base border-r material-icons border-secondary">${
+      const envTypeIcon = `<i class="inline-flex items-center px-1 mr-2 -my-1 text-base border-x material-icons border-secondary">${
         selectedEnvType === "TEAM_ENV" ? "people" : "person"
       }</i>`
+
+      const appendEditAction = (tooltip: HTMLElement) => {
+        const editIcon = document.createElement("span")
+        editIcon.className = "env-icon"
+        editIcon.addEventListener("click", () => {
+          const isPersonalEnv =
+            envName === "Global" || selectedEnvType !== "TEAM_ENV"
+          const action = isPersonalEnv ? "my" : "team"
+          invokeAction(`modals.${action}.environment.edit`, [
+            envName,
+            parsedEnvKey,
+          ])
+        })
+        editIcon.innerHTML = `<i class="inline-flex items-center px-1 ml-2 -my-1 text-base border-x material-icons border-secondary">edit</i>`
```

Comment:
@liyasthomas we still support material icons usage like this ? We downloading the font still ?

Reply:
We should remove material icon font usage. But that is a separate PR, just note this.